pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2364,5 +2364,8 @@ self: super: {
     };
   } super.postgrest));
 
+  # build newer version for `pkgs.shellcheck`
+  ShellCheck_0_9_0 = doDistribute super.ShellCheck_0_9_0;
```

Comment:
I don't think you actually need `doDistribute` here, since it will get built anyway from the top-level `shellcheck` depending on it.

Reply:
It's needed otherwise hydraPlatforms is `[ ]` for `pkgs.shellcheck`.

Reply:
Ah, I see.  So I guess `pkgs.shellcheck` just uses the same hydraPlatforms as `haskellPackages.ShellCheck`.  Makes sense.